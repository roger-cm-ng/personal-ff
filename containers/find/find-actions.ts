import { Dispatch } from 'redux';
import { postToken, postFind } from '../../helpers/resources';
import { FindBody } from './find-types';
import { ERROR_RAISED } from '../error-handler/error-handler-actions';
// eslint-disable-next-line import/no-cycle
import ErrorFindAgain from '../../components/error-content/error-find-again/error-find-again';

export const TOKEN_FETCHED = 'TOKEN_FETCHED';
export const FALCONE_FOUND = 'FALCONE_FOUND';

export const fetchToken = () => async (dispatch: Dispatch) => {
  try {
    const payload = await postToken();

    const { data, status } = payload;

    if (status >= 200 && status < 300) {
      dispatch({
        type: TOKEN_FETCHED,
        payload: data
      });
    } else {
      dispatch({
        type: ERROR_RAISED,
        component: ErrorFindAgain
      });
    }
  } catch (err) {
    dispatch({
      type: ERROR_RAISED,
      component: ErrorFindAgain
    });
  }
};

export const findFalcone = (body: FindBody) => async (dispatch: Dispatch) => {
  try {
    const payload = await postFind(body);

    const { data, status } = payload;

    if (status >= 200 && status < 300) {
      dispatch({
        type: FALCONE_FOUND,
        payload: data
      });
    } else {
      dispatch({
        type: ERROR_RAISED,
        component: ErrorFindAgain
      });
    }
  } catch (err) {
    dispatch({
      type: ERROR_RAISED,
      component: ErrorFindAgain
    });
  }
};
