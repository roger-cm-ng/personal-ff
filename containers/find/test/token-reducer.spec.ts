import tokenReducer from '../token-reducer';
import { TOKEN_FETCHED } from './../find-actions';
import { TOKEN } from './find-mock-data';

describe('token reducer', () => {
  const initialState = () => tokenReducer(null, {
    type: TOKEN_FETCHED,
    payload: null
  });

  const acquiredState = () => tokenReducer(initialState(), {
    type: TOKEN_FETCHED,
    payload: TOKEN
  });

  it('should return initial state', () => {
    expect(initialState()).toEqual(null);
  });

  it('should update the state with acquired data when type is TOKEN_FETCHED', () => {
    expect(acquiredState()).toMatchSnapshot();
    expect(acquiredState()).toEqual(TOKEN);
  });
});