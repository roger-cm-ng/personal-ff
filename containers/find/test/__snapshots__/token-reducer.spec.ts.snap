// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`token reducer should update the state with acquired data when type is TOKEN_FETCHED 1`] = `
Object {
  "token": "12345abcde",
}
`;
